packages/kiota_abstractions/lib/src/serialization/untyped_string.dart

```diff
@@ -0,0 +1,7 @@
+part of '../../kiota_abstractions.dart';
+
+class UntypedString extends UntypedNode {
+  const UntypedString(this.value);
+
+  final String value;
```

Comment:
As far as I can tell from looking at the abstractions for the other languages, the intention seems to be to have that logic in the serialization writer, so I added it there.

Reply:
Okay, I see where you're going with this. Still, in the writer there are these lines in `writeObjectValue`:

```dart
      if (value is UntypedNode) {
        writeUntypedValue(key, value as UntypedNode);
      } else {
        value.serialize(this);
      }
```

`writeUntypedValue` in turn has a switch for more specific types. I'm just wondering why you did this when you could have implemented the `serialize` method from the `Parsable` interface in each type instead. Then you wouldn't have to handle all this in the writer.

Reply:
The UntypedNode and its subclasses represent just the value. I don't think you can implement the serialize method there, because you then lose the key, since the key is not known within the UntypedNode subclasses.

Reply:
Oh I understand now, thanks for explaining! I forgot the `Untyped*` classes live in the abstractions layer. Makes sense that they are independent of the serializer used.